src/JsonLdWriter.php

```diff
@@ -215,10 +228,64 @@ protected function extractPathFromUrl($original)
         return str_replace("/", "", $parts["path"]);
     }
 
-    protected function convertImageUrlToBinary(string $url): ?string
+    protected function convertImageToBinary(array|string $value): mixed
     {
-        // Add code to extract and convert image
+        // First check if we have a simple url string, an array of images or an ImageObject.
+        if (is_string($value)) {
+            // Simple String that we can convert.
+            return $this->convertUrlToBinary($value) ?? $value;
+        } elseif (is_array($value)) {
+            // Either an ImageObject or an Array of images/ImageObjects.
+            if (array_key_exists('@type', $value)) {
+                // ImageObject, we need to find out which field is filled.
```

Comment:
(TBetter to be even more strict here and throw a warning/exception (not sure which is more appropriate) in case `@type!=ImageObject`. 

Schema.org [also defines](https://schema.org/ImageObject) Barcode and ImageObjectSnapshot as subtypes. I guess we do not actually want to extract any of those. Also, we do not want to extract any other weird object that we do not know.